org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/preferences/SonarLintProjectConfigurationManager.java

```diff
@@ -49,14 +54,19 @@ public class SonarLintProjectConfigurationManager {
   public static final String P_BINDING_SUGGESTIONS_DISABLED_KEY = "bindingSuggestionsDisabled";
 
   private static final Set<String> BINDING_RELATED_PROPERTIES = Set.of(P_PROJECT_KEY, P_SERVER_ID, P_BINDING_SUGGESTIONS_DISABLED_KEY);
+  private static Map<ISonarLintProject, Collection<IPreferenceChangeListener>> listenersByScopeContext = new HashMap<>();
```

Comment:
I am a bit skeptical about this map. It sounds like a possible memory leak. Entries are never removed (what if for example a project is closed?).

Reply:
Removed, I don't have to unregister/re-register listeners anymore